tests/integration/it_utils.py

```diff
@@ -302,3 +308,143 @@ def _get_non_decorator_code(function):
+                currency=currency_code,
+                value="1000",
+            ),
+        ),
+        lp_wallet,
+    )
+
+    sign_and_reliable_submission(
+        Payment(
+            account=issuer_wallet.classic_address,
+            destination=lp_wallet.classic_address,
+            amount=IssuedCurrencyAmount(
+                currency=currency_code,
+                issuer=issuer_wallet.classic_address,
+                value="500",
+            ),
+        ),
+        issuer_wallet,
+    )
+
+    sign_and_reliable_submission(
+        AMMCreate(
+            account=lp_wallet.classic_address,
+            amount="250",
+            amount2=IssuedCurrencyAmount(
+                issuer=issuer_wallet.classic_address,
+                currency=currency_code,
+                value="250",
+            ),
+            trading_fee=12,
+        ),
+        lp_wallet,
+        client,
+    )
+
+    return {
+        "issuer_wallet": issuer_wallet,
+        "lp_wallet": lp_wallet,
+        "asset": XRP(),
+        "asset2": IssuedCurrency(
+            currency=currency_code,
+            issuer=issuer_wallet.classic_address,
+        ),
+    }
+
+
+async def setup_amm_pool_async(
+    client: AsyncClient = ASYNC_JSON_RPC_CLIENT,
+) -> Dict[str, Any]:
+    issuer_wallet = Wallet.create()
+    await fund_wallet_async(issuer_wallet)
+    lp_wallet = Wallet.create()
+    await fund_wallet_async(lp_wallet)
+    currency_code = "USD"
+
+    # test prerequisites - create trustline and send funds
+    await sign_and_reliable_submission_async(
+        AccountSet(
+            account=issuer_wallet.classic_address,
+            set_flag=AccountSetAsfFlag.ASF_DEFAULT_RIPPLE,
+        ),
+        issuer_wallet,
+    )
+
+    await sign_and_reliable_submission_async(
+        TrustSet(
+            account=lp_wallet.classic_address,
+            flags=TrustSetFlag.TF_CLEAR_NO_RIPPLE,
+            limit_amount=IssuedCurrencyAmount(
+                issuer=issuer_wallet.classic_address,
+                currency=currency_code,
+                value="1000",
+            ),
+        ),
+        lp_wallet,
+    )
+
+    await sign_and_reliable_submission_async(
+        Payment(
+            account=issuer_wallet.classic_address,
+            destination=lp_wallet.classic_address,
+            amount=IssuedCurrencyAmount(
+                currency=currency_code,
+                issuer=issuer_wallet.classic_address,
+                value="500",
+            ),
+        ),
+        issuer_wallet,
+    )
+
+    await sign_and_reliable_submission_async(
+        AMMCreate(
+            account=lp_wallet.classic_address,
+            amount="250",
+            amount2=IssuedCurrencyAmount(
+                issuer=issuer_wallet.classic_address,
+                currency=currency_code,
+                value="250",
+            ),
+            trading_fee=12,
+        ),
+        lp_wallet,
+        client,
+    )
+
+    return {
```

Comment:
Why not return an `AMM` ledger object, to match the return shape of all the other reusable values?

Reply:
I decided to remove `AMM` so it doesn't get mistaken as AMM ledger object. Now, I just returned 3 variables that are needed for tests: `AMM_ASSET`, `AMM_ASSET2`, `AMM_ISSUER_WALLET`.